rllib/execution/parallel_requests.py

```diff
@@ -182,3 +183,224 @@ def wait_asynchronous_requests(
         remote_fn=None,
     )
     return ready_requests
+
+
+class AsyncRequestsManager:
+    """A manager for asynchronous requests to actors.
+
+    Args:
+        workers: A list of ray remote workers to operate on. These workers must have an
+            `apply` method which takes a function and a list of arguments to that
+            function.
+        max_remote_requests_in_flight_per_actor: The maximum number of remote
+            requests that can be in flight per actor. Any requests made to the pool
+            that cannot be scheduled because the
+            max_remote_requests_in_flight_per_actor per actor has been reached will
+            be queued.
+        ray_wait_timeout_s: The maximum amount of time to wait for inflight requests
+            to be done and ready when calling
+            AsyncRequestsManager.get_ready_requests().
+
+    Example:
+        >>> import time
+        >>> import ray
+        >>> from ray.rllib.execution.parallel_requests_manager import (
+        ...     AsyncRequestsManager)
+        >>>
+        >>> @ray.remote
+        ... class MyActor:
+        ...    def apply(self, fn, *args: List[Any], **kwargs: Dict[str, Any]) -> Any:
+        ...        return fn(*args, **kwargs)
+        ...
+        ...    def task(self, a: int, b: int) -> Any:
+        ...        time.sleep(0.5)
+        ...        return a + b
+        >>>
+        >>> workers = [MyActor.remote() for _ in range(3)]
+        >>> manager = AsyncRequestsManager(workers,
+        ...                                max_remote_requests_in_flight_per_actor=2)
+        >>> manager.submit(lambda worker, a, b: worker.task(a, b), fn_args=[1, 2])
+        >>> print(manager.get_ready_requests())
+        >>> manager.submit(lambda worker, a, b: worker.task(a, b),
+        ...                fn_kwargs={"a": 1, "b": 2})
+        >>> time.sleep(2) # Wait for the tasks to finish.
+        >>> print(manager.get_ready_requests())
+    """
+
+    def __init__(
+        self,
+        workers: List[ActorHandle],
+        max_remote_requests_in_flight: int = 2,
+        ray_wait_timeout_s: Optional[float] = 0.03,
+        return_object_refs: bool = False,
+    ):
+        self._ray_wait_timeout_s = ray_wait_timeout_s
+        self._return_object_refs = return_object_refs
+        self._max_remote_requests_in_flight = max_remote_requests_in_flight
+        self._all_workers = set(workers)
+        self._num_workers = len(workers)
+        self._available_workers = workers.copy()
+        self._available_workers_set = set(self._available_workers)
```

Comment:
why do you keep both, _available_workers and _available_workers_set?
keep them both in sync is additional work, no?